### PR TITLE
Fix login route

### PR DESCRIPTION
**Pull request by nLight**
As being said… Routes are super confusing, so I broke them while trying to fix.

This PR addresses all the cases:
- normal routes
- plugins
- 404
- login
- logout
- open cluster

🙇

### CODE REVIEW

**Comment by nLight** [comment]
retest this please 😩

**Comment by nLight** [comment]
retest this please 🌧

File: src/js/routes/index.js [hunk 1]
```diff
@@ -55,21 +45,29 @@ function getApplicationRoutes() {
...
-          // making all the routes going after this point unreachable
+          // This is a bit tricky
+          // NotFoundPage should not be among children of id: index
+          // yet it expects Index to be its parent
           type: Route,
-          path: '*',
-          component: NotFoundPage
+          component: Index,
+          children: [
+            // Please make sure this is THE LAST route and is not inside of id: 'index'
```
Comment (orlandohohmeier) [inline src/js/routes/index.js]: Please update this comment

**Comment by wavesoft** [comment]
Looks good to me ✨

File: src/js/routes/index.js [hunk 2]
```diff
@@ -53,15 +53,21 @@ function getApplicationRoutes() {
...
+          // `NotFoundPage` should not be among children of `id: index` route
+          // because `path: '*'` is a glob and matches everything
+          // so it will prevent all the routes that go after from being resolved
+          // in our case these will be plugins routes
+          //
+          // Yet it expects Index to be its parent,
+          // this is why we have `component: Index` twice
+          // I decided not to add component Index inside NotFoundPage
+          // to make it explicit
+          // TODO: make it less tricky
```
Comment (orlandohohmeier) [inline src/js/routes/index.js]: Let's create a JIRA so that we don't forget

Comment (nLight) [inline src/js/routes/index.js]: 👍

**Comment by nLight** [comment]
@orlandohohmeier addressed 👍

**Comment by orlandohohmeier** [comment]
✨ Awesome

**Pull request merged**